crates/python/src/lib.rs

```diff
@@ -17,15 +18,32 @@ create_exception!(qcs, CompilationError, PyRuntimeError);
 create_exception!(qcs, RewriteArithmeticError, PyRuntimeError);
 create_exception!(qcs, DeviceIsaError, PyValueError);
 
-#[pyfunction]
-fn compile(py: Python<'_>, quil: String, target_device: String) -> PyResult<&PyAny> {
+#[pyfunction(kwds = "**")]
+fn compile<'a>(
+    py: Python<'a>,
+    quil: String,
+    target_device: String,
+    kwds: Option<&PyDict>,
+) -> PyResult<&'a PyAny> {
     let target_device: TargetDevice =
         serde_json::from_str(&target_device).map_err(|e| DeviceIsaError::new_err(e.to_string()))?;
+
+    let mut compiler_timeout = Some(DEFAULT_COMPILER_TIMEOUT);
```

Comment:
Depending on how you handle the duplication of `DEFAULT_COMPILER_TIMEOUT` and `DEFAULT_CLIENT_TIMEOUT`, you may want to change this.